libexec/python/docker/api.py

```diff
@@ -47,24 +52,46 @@ def create_runscript(cmd,base_dir):
     return output_file
 
 
-def get_token(repo_name,namespace="library",scope="repository",permission="pull"):
+def get_token(registry,repo_name,namespace,scope="repository",permission="pull",auth=None):
     '''get_token will use version 2.0 of Docker's service to return a token with given permission and scope - this
```

Comment:
Do you mean having get_token be provided at all? For organization - I don't see any reason that someone might want to use it, you are right that it might be more appropriate as a utility or authentication function - if we had a larger code base it would be appropriate to have an auth.py in the docker folder, but I didn't think this level of organization was necessary (yet anyway).  For what my comment was about - Is there a reason for not having the registry default to index.docker.io as we do in the other functions? That should be easy to add, even it's not a complete function (eg, the same three lines of code are repeated in multiple places in the file!)